src/scripts/main.js

```diff
@@ -1 +1,56 @@
 'use strict';
+
+function showNotification(message, type) {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.className = type;
+  notification.textContent = message;
+  document.body.appendChild(notification);
+}
+
+const firstPromise = new Promise(function (resolve, reject) {
+  document.addEventListener('click', () => {
+    clearTimeout(timer);
```

Comment:
The variable 'timer' is used before it is declared. Consider moving the declaration of 'timer' before the 'click' event listener to avoid potential reference errors.